docs/eBpfExtensions.md

```diff
@@ -252,15 +252,15 @@ There is only one function in the client dispatch table, which is of the followi
 
 ```
 /**
- *  @brief This is the only function in the eBPF Hook NPI client dispatch table.
+ *  @brief This is the only mandatory function in the eBPF Hook NPI client dispatch table.
```

Comment:
Since this dispatch table is provided by the client (ebpfcore), all the functions are mandatory? I think it is optional for the extensions to use all these functions, but they will be there.

Reply:
We may also need to update line 251 as it states  `There is only one function in the client dispatch table ...`

Reply:
The contract is actually not entirely clear to me; in some of eBPF's test code, only the program_invoke function is plumbed and the rest are empty. It would be simplest for extensions if all functions are guaranteed to be supplied by eBPF. @Alan-Jowett was that the intent when batched inspection was added?

Reply:
I've updated the documentation to remove the "only mandatory" qualification and tidied it up a bit, but I'm not knowledgeable enough to specify the contract. Should I open a bug to track this? We're also missing documentation for all the batched routines.

Reply:
I've referenced the missing documentation in #2331.